// @shuddho11288/blackboxai-api/index.js
const axios = require("axios");
/**
 * 
 * @function blackbox - sends a prompt to blackbox and returns the response
 * @param {String} prompt - prompt to be sent to blackbox
 * @param {Boolean} web_search - set to true if you want to use web search. By default it is set to false.
 * @returns 
 */
const blackbox = async (prompt, web_search = false) => {
    
  const body = {
    messages: [
      {
        id: "aX0Smmm",
        content: prompt,
        role: "user"
      }
    ],
    id: "aX0Smmm",
    previewToken: null,
    userId: null,
    codeModelMode: true,
    agentMode: {},
    trendingAgentMode: {},
    isMicMode: false,
    userSystemPrompt: null,
    maxTokens: 1024,
    playgroundTopP: 0.9,
    playgroundTemperature: 0.5,
    isChromeExt: false,
    githubToken: "",
    clickedAnswer2: false,
    clickedAnswer3: false,
    clickedForceWebSearch: false,
    visitFromDelta: false,
    mobileClient: false,
    userSelectedModel: null,
    validated: "00f37b34-a166-4efb-bce5-1312d87f2f94",
    imageGenerationMode: false,
    webSearchModePrompt: false
  };
  
  const config = {
    headers: {
      'Content-Type': 'application/json',
      'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.6778.86 Safari/537.36',
      'Accept-Language': 'en-US,en;q=0.9',
      'Origin': 'https://www.blackbox.ai',
      'Referer': 'https://www.blackbox.ai',
      'Cookie': 'sessionId=ac7174d6-9eef-4442-a135-829c4afac237; __Host-authjs.csrf-token=cbee3a973bd3039faa96360bf1ad84faab8e9e97a204e960098dd00a5031570b%7C31a607aa82b17e430f1c1f720c8e21dc7fe15ff52944f53b18fe415eceb73853; __Secure-authjs.callback-url=https%3A%2F%2Fwww.blackbox.ai'
    }
  };
  
  const url = "https://www.blackbox.ai/api/chat";
  
  const response = await axios.post(url, body, config)

  return response.data
  
}

// blackbox('hello').then(data=>{
//     console.log(data)
// })

module.exports = {
    blackbox
}
